### DISCUSSION TITLE
IBOS Q&A Miscellaneous

### DISCUSSION THREAD

**Discussion by fromITzerotoIThero** [fromITzerotoIThero/IBOS#26] [Announcements]
Dear W., 

welcome to IBOS Discussions 😉 !

Sometime ago I wrote some code of Cart Indicator + Redux, on branch `feature/19-shopping-cart-indicator-redux`.
The branch is up-to-date with its remote, but it's far from where master branch is now. 
What I think I should do, is delete it and create an entirely new branch stemming from the latest version of master branch.
I wanted to make sure I got it right :)

**Comment by fromITzerotoIThero** [comment]
W., 

What happens when I decide to run some more changes in the code on a branch after I make a PR? I suppose the version you get to review is the one with the very latest changes, but wanted to make sure, so as not to screw things up XD

**Comment by piotrwicijowski** [comment]
As long as you have not finished the PR then it is all fine - that's how you would apply changes to any comments during the review. But once the PR is completed and merged - please do not re-use the branch (the best practice is to delete the branch during merging)

**Reply by fromITzerotoIThero** [reply]
👌 Thank you!

**Comment by fromITzerotoIThero** [comment]
Oh W., 

I've just noticed a weird routing behavior on deployed[ IBOS website](https://ibos.netlify.app):

- when I navigate to pages other than `home`, and refresh it (both with F5 and browsers refresh button), I get 404 error
- back and forward navigation works as expected

Where should I look for the source of the bug - is that a routing issue in the app code or it's due to some netlify configuration?

**Comment by piotrwicijowski** [comment]
That's netlify configuration - you would have to tell it that no matter which url your browser goes to, the server should send the index.html file (since we have SPA - single page application). For more details see here https://www.netlify.com/blog/2020/04/07/creating-better-more-predictable-redirect-rules-for-spas/ - in short creating a _redirects file should be enough

**Reply by fromITzerotoIThero** [reply]
Thank you! :)

I have another issue though, prompted by my latest PR (the one deleting extra ImageCarousel), and it concerns working on real life projects.

When I'm working on a feature on a branch with someone, before I get to code I should:
- merge master into that feature branch (in case there are some changes made by other part of the team;
- pull for that feature branch from remote in case the person I'm working with on a feature made some changes?

Did I get it right?
Or maybe I should do the merging-master part only before making a PR?

**Reply by fromITzerotoIThero** [reply]
On this netlify configuration issue - I seem to know what to do in order to fix it, but I haven't a faintest idea how to test it locally prior to making a PR and merging to master - the problem is not there when working on development server... 
Should I make code changes, make a PR, then merge it and only then see if it helped address the issue? 
Or am I missing something?

**Reply by piotrwicijowski** [reply]
That's true, there is no way to test it just locally. However, if you make a PR, it then gets automatically deployed to a preview environment (maybe you have noticed that), and there, even before blind-merging you can see the results. And if it does not work on the first try - you can make changes on the branch and push them, and the netlify integration with the PR should detect that and update the preview environment.

**Reply by fromITzerotoIThero** [reply]
Oh, you're right, there was that option, but I completely forgot about it! I'm gonna try it out :)

**Comment by piotrwicijowski** [comment]
I highly recommend **not** to work with someone on the same branch, because it can lead to a lot of headaches whenever either of you would forget to pull the latest commits from remote. But yeah, if you are forced to do this, then the point "pull for that feature branch from remote in case the person I'm working with on a feature made some changes" is a very important one.

And when it comes to merging master back into the feature branch - it is very rarely needed, only maybe when you get some hard merge conflicts, or the feature branch is very very old and you want to make sure on your local that the changes would not break anything. But aside from those edge cases, most of the time you can do without these back-merges.

**Comment by fromITzerotoIThero** [comment]
W., 

I have 2 more questions:
1. I've just made a new branch for `ProductList` pagination, but:
- I screwed the name and I deleted it locally
- yet I forgot I've already published it 
- (no changes saved or commits made) - it's like an empty branch
- how should I delete the remote branch now? 
- GitHub suggests that I make a PR

2. How do you think it's best to manage these Discussions? Should I create a separate Discussion for each topic? That sounds reasonable, but maybe it's not needed, what's your take on it?

**Reply by piotrwicijowski** [reply]
1. You can delete a remote branch through the github page -> go to code -> branches -> find your branch -> click on the trash icon on the right
2. For short term this is fine for me, but managing the discussions can be interesting depending on how do we want to use them in the long term. Since a lot of those discussions are not only one-time quick fixes, but more general discussions about git management, code quality etc, and in case you want to reference them in the future to remind yourself on how things were solved and why, then I guess these can be split up to those topics. Then it can become a growing reference guide for you, e.g. "I remember I had a weird issue with git branches back in the day" - it would be easier to sift through a topic specific discussion rather than this whole catch-all bucket
